examples/async_stream/main.cc

```diff
@@ -0,0 +1,32 @@
+#include <drogon/drogon.h>
+#include <chrono>
+#include <memory>
+using namespace drogon;
+using namespace std::chrono_literals;
+
+int main()
+{
+    app().registerHandler(
+        "/stream",
+        [](const HttpRequestPtr &,
+           std::function<void(const HttpResponsePtr &)> &&callback) {
+            auto resp = drogon::HttpResponse::newAsyncStreamResponse(
+                [](drogon::StreamPtr stream) {
+                    std::thread([stream = std::shared_ptr<drogon::Stream>{
```

Comment:
why not use `stream = std::move(stream)` ?

Reply:
The lambda might be copied around. Wrapping the unique_ptr in a shared_ptr should prevent any attempt in copying the std::unique_ptr in all cases.